File: twosum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const oldvalues ={}
    for (let i = 0; i < nums.length; i++) {
        const CV = nums[i]
        const NV = target - CV
        const index2 = oldvalues[NV]
        if (index2 != null) {
            return [index2, i]
        } else {
            oldvalues[CV] = i
        }
    }
};

File: twosum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        prevalues = {}
        
        for i, n in enumerate(nums):
            opp = target - n
            if opp in prevalues: 
                return [prevalues[opp], i]
            prevalues[n] = i

        
